### PR TITLE
refactoring wip

### PR DESCRIPTION
**Pull request by adaojunior**
Sorry there's a lot of changes :/

### CODE REVIEW

**Comment by ntaoo** [comment]
Thank you for the PR, but sorry at first could you extract sass stuff from other changes and create the new issue for sass?
I'm also working on demo app enhancements right now, so I want to merge your PR code on demo app with my working demo_app repo.

**Comment by ntaoo** [comment]
Let's follow the package convention.
> (You may be asking whether you should put your web-based example programs in example or web? Put those in example.)
> https://www.dartlang.org/tools/pub/package-layout.html

**Comment by adaojunior** [comment]
@ntaoo https://pub.dartlang.org/packages/peanut

File: lib/components.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const MD_DIRECTIVES = const [
+  MD_BUTTON_DIRECTIVES,
+  ///MD_CARD_DIRECTIVES,
+  MD_ICON_DIRECTIVES,
+  MD_LIST_DIRECTIVES,
+  MD_SIDENAV_DIRECTIVES,
+  MD_TOOLBAR_DIRECTIVES,
+  Dir
+];
```
Comment (ntaoo) [inline lib/components.dart]: @adaojunior Let's wait for the progress below.
https://github.com/angular/material2/issues/236

My impression is that all in one components helper would be useful only on demo app, and mixing different types of the component into one directive might not a good idea on production app.

Once we add our original features, even this helper looks thin and convenient at present, we have to maintain it and would eventually mark deprecate metadata, then removing.

I want to keep side by side porting work as much as we can at current phase.

**Pull request merged**